### PR TITLE
[dualtor][orchagent] add 'test_orch_slb'

### LINKED ISSUE #3278: SLB Test Cases
Step | Goal | Expected results
-- | -- | --
Create peering session from the SLB to Active ToR | SLB | Verify session is established
Create peering session from the SLB to Standby ToR | SLB | Verify session is established
  |   |  
Announce routes from SLB to Active ToR | SLB | Verify routes in Active ToR
Announce routes from SLB to Standby ToR | SLB | Verify routes in Standby ToR
  |   |  
Run PTF tests on Active ToR | SLB | Verify packets forwarded directly to active SLB port
Run PTF tests on Standby ToR | SLB | Verify packets forwarded via tunnel to Active ToR
  |   |  
Withdraw routes from SLB to Active ToR | SLB | Verify routes removed in Active ToR
Withdraw routes from SLB to Standby ToR | SLB | Verify routes removed in Standby ToR
  |   |  
Repeat PTF tests as above | SLB | Verify no packets forwarded
  |   |  
Simulate a mux state change for the SLB port | SLB | Verify both sessions stays established and not disrupted
  |   |  
Announce routes from SLB to new Active ToR | SLB | Verify routes in Active ToR
Announce routes from SLB to new Standby ToR | SLB | Verify routes in Standby ToR
  |   |  
Repeat PTF tests as above | SLB | Verify packet forwarding based on mux state
  |   |  
Verify teardown by shutting peering session one by one | SLB | After one session is down, verify other peering session is active and routes present

### PR DESCRIPTION
**Pull request by lolyu**
### Description of PR

Add `test_orch_slb` to cover the SLB scenario.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

Summary:
Fixes #3278

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Test SLB under dualtor scenario.

#### How did you do it?
1. Put the BGP neighbor class in a shareable module.
2. Add `test_orch_slb` to start two bgp sessions separately to both active and standby ToR, verify bgp session establishment, route announcement and withdrawal, and packet forwarding.

#### How did you verify/test it?
Run the test.

### CODE REVIEW

**Comment by lolyu** [comment]
```
dualtor/test_orchagent_slb.py::test_orchagent_slb PASSED                                                                                                                                       [100%]

===================================================================================== 1 passed in 232.25 seconds =====================================================================================
```

File: tests/dualtor/test_orchagent_slb.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            conn["exabgp_port"],
+            conn["neighbor_type"],
+            is_quagga=is_quagga
+        )
+    return neighbors
+
+
+@pytest.fixture
+def constants(setup_interfaces):
+    class _C(object):
```
Comment (theasianpianist) [inline tests/dualtor/test_orchagent_slb.py:154]: Why do we need to create a dummy class as opposed to using a dictionary to store this info?

Comment (lolyu) [reply tests/dualtor/test_orchagent_slb.py:154]: Just prefer using '.' usage.

**Review by theasianpianist — APPROVED** [review]
lgtm. @prsunny any comments?

**Pull request merged**